python/ray/autoscaler/batching_node_provider.py

```diff
@@ -230,6 +243,16 @@ def terminate_node(self, node_id: str) -> Optional[Dict[str, Any]]:
                 f"{node_type}. Skipping termination request."
             )
 
-        self.scale_request.desired_num_workers[node_type] -= 1
-        self.scale_request.workers_to_delete.add(node_id)
+        # Scale down entire replica if part of a TPU podslice being deleted
+        node_replica_index = self.node_data_dict[node_id].replica_index
+        if node_replica_index:
+            for worker in self.replicas_to_nodes[node_replica_index]:
+                # Check if node has already been scheduled to delete
+                if node_id not in self.scale_request.workers_to_delete:
+                    # Assume all workers in a group are of the same type
+                    self.scale_request.desired_num_workers[node_type] -= 1
+                    self.scale_request.workers_to_delete.add(node_id)
```

Comment:
```suggestion
                    self.scale_request.workers_to_delete.add(worker)
```

Reply:
I changed the var to `worker_id` for consistency in 96272bb.